framework/filters/AccessRule.php

```diff
@@ -161,8 +191,13 @@ protected function matchRole($user)
                 if (!$user->getIsGuest()) {
                     return true;
                 }
-            } elseif ($user->can($role)) {
-                return true;
+            } else {
+                if (!isset($roleParams)) {
```

Comment:
and a bit off-topic but i can't load the original commit.

can you edit line 148?

`return $this->allow ? true : false;`

`$allow` is bool so the ternary is unneeded.

```php
public $allow = false; // currently is null which gets taken as false
// ...
return $this->allow;
```php

or

`return (bool) $this->allow; // null case remains identical to 'false'`